conda_smithy/validate_schema.py

```diff
@@ -37,7 +44,20 @@ def get_validator_class():
 
 
 def validate_json_schema(
-    config, schema_file: str = None
+    config: Union[
+        CommentedMap,
+        Dict[str, str],
+        Dict[str, Union[str, List[str]]],
+        Dict[
+            str,
+            Union[
+                Dict[str, Union[bool, Dict[str, str]]],
+                Dict[str, Dict[str, str]],
+            ],
+        ],
+        Dict[str, Union[str, bool]],
+    ],
```

Comment:
I believe this type is equivalent to:

    Union[
        CommentedMap,
        Dict[str, bool],
        Dict[str, str],
        Dict[str, List[str]],
        Dict[str, Dict[str, bool]],
        Dict[str, Dict[str, Dict[str, str]]]
    ]

Or, if you want to remove a bit of duplication without too much extra union nesting:

    Union[
        CommentedMap,
        Dict[str, Union[
            bool,
            str,
            List[str],
            Dict[str, bool],
            Dict[str, Dict[str, str]]
        ]]
    ]

I think either of these is more readable than the existing type.